src/app/chat/view.nim

```diff
@@ -264,17 +270,37 @@ QtObject:
     self.status.chat.chatMessages(self.activeChannel.id, false)
     self.messagesLoaded();
 
+  proc loadMoreMessagesWithIndex*(self: ChatsView, channelIndex: int) {.slot.} =
+    if (self.chats.chats.len == 0): return
+    let selectedChannel = self.chats.getChannel(channelIndex)
+    if (selectedChannel == nil): return
+    trace "Loading more messages", chaId = selectedChannel.id
+    self.status.chat.chatMessages(selectedChannel.id, false)
+    self.messagesLoaded();
```

Comment:
This function will probably need to change in a separate PR. The "Fetch Messages" functionality is for requesting messages from the mailserver, while `chatMessages` is for loading messages already stored in status-go